torch/_refs/__init__.py

```diff
@@ -3565,12 +3555,21 @@ def transpose(a: TensorLikeType, dim0: int, dim1: int) -> TensorLikeType:
 swap_axes = transpose
 
 
-@register_decomposition(torch.ops.aten.unfold_copy)
-def unfold_copy(a: TensorLikeType, dimension: int, size: int, step: int):
-    new_size, new_stride = _get_unfold_copy_shape_stride(
-        a.shape, a.stride(), dimension, size, step
+@register_decomposition(torch.ops.aten.unfold)
+def unfold(
+    self: TensorLikeType, dimension: int, size: int, step: int
+) -> TensorLikeType:
+    shape, strides = _get_unfold_shape_stride(
+        self.shape, self.stride(), dimension, size, step
     )
-    return a.as_strided(new_size, new_stride)
+    return self.as_strided(shape, strides)
+
+
+@register_decomposition(torch.ops.aten.unfold_copy)
+@out_wrapper()
+def unfold_copy(self: TensorLikeType, dimension: int, size: int, step: int):
+    # nb. It's not clear to me what's the difference between this function and torch.unfold
```

Comment:
unfold_copy returns a contiguous tensor with the result, whereas `unfold` returns restrided view of the input. It's unfortunate view tests no longer catch those discrepancies.

Reply:
The signatures in native_functions.yaml show this aliasing relationship:

https://github.com/pytorch/pytorch/blob/29c78266c046c7f83e7d84fc764af47e62ae9542/aten/src/ATen/native/native_functions.yaml#L8881

https://github.com/pytorch/pytorch/blob/29c78266c046c7f83e7d84fc764af47e62ae9542/aten/src/ATen/native/native_functions.yaml#L12925

There is a test that unfold returns a view (at least in one case):

https://github.com/pytorch/pytorch/blob/0b93afb112d48bb6d89a1e183a90b403560e84e4/test/test_view_ops.py#L602

```
t = torch.ones(10)
v = t.unfold(0, 3, 2)
v._is_view()
: True

t = torch.ones(10)
c = torch.unfold_copy(t, 0, 3, 2)
c._is_view()
: False
```

Reply:
Right, there are one-off tests here and there, but in general we've given up on testing that refs preserve view relationships of original ops, and refs won't be tested on those one-off tests.

Reply:
Correct, sorry, didn't mean to suggest otherwise

Reply:
> Ok, so hear this out. It so happens that unfold_copy does return a view lol

Do you have a repro? The impl for the view copy kernels lives in `build/aten/src/ATen/CompositeViewCopyKernel.cpp` (codegen), and looks like this:
```
at::Tensor unfold_copy(const at::Tensor & self, int64_t dimension, int64_t size, int64_t step) {
  auto output = at::_ops::unfold::call(self, dimension, size, step);
    return output.clone();
}
```

The output isn't a view (since we call `.clone()`). Although the output should be contiguous, as Natalia mentioned above, and that technically isn't always the case. That should be fixed by https://github.com/pytorch/pytorch/pull/85747.

What's the failing out= test? My guess is that the out= kernels aren't resizing properly (they're implemented today by just calling the original `unfold_copy()` op, followed by a `copy_()` to the out= tensor).

Reply:
I don't think unfold_copy produces a view, except in rare circumstances where unfold is a no-op:
```
In [3]: x = torch.arange(1., 8)

In [4]: x.data_ptr()
Out[4]: 93952253159872

In [5]: x.unfold(0, 2, 1).data_ptr()
Out[5]: 93952253159872

In [6]: x.unfold(0, 2, 1).is_contiguous()
Out[6]: False

In [11]: out=torch.unfold_copy(x, 0, 2, 1)

In [12]: out.data_ptr()
Out[12]: 93952337762432

In [13]: out.is_contiguous()
Out[13]: True
```

Reply:
oof. Sorry for the noise. I should have done an unconditional copy, but `contiguous()` sometime returns an alias. That's fixed.

Now, the `out=` errors are real. I have not investigated what's it that's failing exactly.